packages/react-components/react-avatar/src/utils/partitionAvatarGroupItems.ts

```diff
@@ -4,29 +4,37 @@ export type PartitionAvatarGroupItemsOptions<T> = {
   maxInlineItems?: number;
 };
 
+export type PartitionAvatarGroupItems<T> = {
+  inlineItems: readonly T[];
+  overflowItems?: readonly T[];
+};
+
 /**
  * Get the inline items and overflowing items based on the array of AvatarGroupItems needed for AvatarGroup.
  *
  * @param options - Configure the partition options
  *
  * @returns Two arrays split into inline items and overflow items based on maxInlineItems.
  */
-export const partitionAvatarGroupItems = <T>(options: PartitionAvatarGroupItemsOptions<T>) => {
+export const partitionAvatarGroupItems = <T>(
+  options: PartitionAvatarGroupItemsOptions<T>,
+): PartitionAvatarGroupItems<T> => {
   const { items } = options;
   const isPie = options.layout === 'pie';
 
   if (isPie) {
     return {
       inlineItems: items.slice(0, 3),
-      overflowItems: items,
+      overflowItems: items.length > 3 ? items : undefined,
```

Comment:
Can you explain this logic ? I would expect `overflowItems` to contain the items that actually overflow, why does it contain **all** items when there are more than 3 ?